test/integration/authorize/basic_test.rb

```diff
@@ -657,6 +657,54 @@ def setup
 
       assert_not_authorized
       assertions.call false
+
+      # Test that hitting children over the parent limit does not translate to
+      # the parent so that the call is still authorized.
+      get '/transactions/authorize.xml', {
+          :provider_key => @provider_key,
+          :app_id       => application.id,
+          :usage        => { metric_child2 => parent_limit + 1 },
+        },
+        'HTTP_3SCALE_OPTIONS' => Extensions::FLAT_USAGE
+      Resque.run!
+
+      assert_authorized
+      assertions.call false
+      #
+      get '/transactions/authorize.xml', {
+          :provider_key => @provider_key,
+          :app_id       => application.id,
+          :usage        => { metric_child1 => parent_limit + 1,
+                             metric_child2 => parent_limit + 1 },
+        },
+        'HTTP_3SCALE_OPTIONS' => Extensions::FLAT_USAGE
+      Resque.run!
+
+      assert_authorized
+      assertions.call false
+
+      # Using flat usage still goes over the limits of specified metrics
+      get '/transactions/authorize.xml', {
+          :provider_key => @provider_key,
+          :app_id       => application.id,
+          :usage        => { parent => parent_limit + 1 },
+        },
+        'HTTP_3SCALE_OPTIONS' => Extensions::FLAT_USAGE
+      Resque.run!
+
+      assert_not_authorized
+      assertions.call false
+
+      get '/transactions/authorize.xml', {
+          :provider_key => @provider_key,
+          :app_id       => application.id,
+          :usage        => { metric_child1 => parent_limit + 1 + 1 },
```

Comment:
Yes, that's better. :+1: However we are depending in the tests above on `child1_limit` being higher than its parent's, perhaps we should encode that in an assertion too, in case we change the values.